static/src/javascripts/lib/storage.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+/* eslint consistent-return: 0, no-plusplus: 0 */
+
+class Storage {
+    storage: any;
+    available: ?boolean;
+
+    constructor(type: string) {
+        this.storage = window[type];
+        this.available = undefined;
+    }
+
+    isAvailable(): ?boolean {
```

Comment:
It is worth considering whether this whole function is still necessary. Our supported browsers don't go as far back as iOS 5 and I don't think the issue outline in SO is still happening.